### PR TITLE
BREAKING CHANGE: make strictQuery false by default again

### LINKED ISSUE #11861: Make `strictQuery` `false` by default again
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Mongoose version

6.x

### Node.js version

Any

### MongoDB server version

Any

### Description

The `strictQuery` change in Mongoose 6 (https://mongoosejs.com/docs/migrating_to_6.html#strictquery-is-removed-and-replaced-by-strict) has caused a lot of confusion. Here's a couple:

https://github.com/Automattic/mongoose/issues/11609
https://github.com/Automattic/mongoose/issues/11807
https://github.com/Automattic/mongoose/issues/11514

Etc.

### Steps to Reproduce

From #11609: 

```javascript
const mongoose = require('mongoose');

const testSchema = new mongoose.Schema({
    title: [{
        value: String,
        language: String
    }]
});

const Test = mongoose.model('Test', testSchema);

async function run() {
    await mongoose.connect('mongodb://localhost:27017');
    await mongoose.connection.dropDatabase();

    await Test.create({
        title: [{
            value: 'El Dave valencia',
            language: 'ca'
        }, {
            value: 'El Dave castellano',
            language: 'es'
        }]
    });

    await Test.create({
        title: [{
            value: 'test',
            language: 'ca'
        }, {
            value: 'test',
            language: 'es'
        }]
    });

    const query = await Test.find({$or: [{"title.value":/Dave/i},{"name.value":/Dave/i} ]});
    console.log('done', query, query[0].title, query[1].title);
    
}

run();
```

### Expected Behavior

Make `strictQuery` false by default in Mongoose 7

### PR DESCRIPTION
**Pull request by vkarpov15**
Fix #11861

**Summary**

Finally :+1:

**Examples**

### CODE REVIEW

File: docs/guide.md [hunk 1]
```diff
@@ -882,19 +882,9 @@ const docs = await MyModel.find(req.query);
...
-// query filter properties by default.
-// This does **not** affect `strict`.
-mongoose.set('strictQuery', false);
-```
-
-In Mongoose 7, `strictQuery` default value will be switched back to `false`.
-You can prepare for the change by specifying:
-
 ```javascript
 // Set `strictQuery` to `false` to prepare for the change
```
Comment (hasezoey) [inline docs/guide.md:889]: i think this comment can also be removed now

File: lib/cast.js [hunk 2]
```diff
@@ -378,26 +378,18 @@ function getStrictQuery(queryOptions, schemaUserProvidedOptions, schemaOptions,
   if ('strictQuery' in queryOptions) {
     return queryOptions.strictQuery;
   }
-  if ('strict' in queryOptions) {
```
Comment (hasezoey) [inline lib/cast.js:381]: why is this removed? is it meant so that `strict` does not affect `strictQuery` anymore?

**Review by hasezoey — COMMENTED** [review]
mostly looks good to me, though some questions / notes

File: lib/cast.js (hunk 2 above)
Comment (vkarpov15) [reply lib/cast.js:381]: Yes exactly. We made `strict` tied to `strictQuery` as a workaround for the `strictQuery: true` by default issue. Now that should no longer be necessary.

**Review by hasezoey — APPROVED** [review]
LGTM (the extra unrelated commits make it hard to review again)

**Comment by vkarpov15** [comment]
Yeah not sure why those extra changes got pulled in. I merged master into 7.0... Weird.
